src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/PaymentBuilderWidget.tsx

```diff
@@ -24,7 +25,12 @@ import { ExpenditureStep, type PaymentBuilderWidgetProps } from './types.ts';
 import { getCancelStepIndex, getExpenditureStep } from './utils.ts';
 
 const PaymentBuilderWidget: FC<PaymentBuilderWidgetProps> = ({ action }) => {
-  const { colony } = useColonyContext();
+  const { colony, refetchColony } = useColonyContext();
+  const { refetch: refetchExpenditures } = useGetColonyExpendituresQuery({
+    variables: {
+      colonyAddress: colony.colonyAddress,
+    },
+  });
```

Comment:
Would it also work if you changed the fetch policy of that query inside `useColonyExpenditureBalances` to something like `cache-and-network`? That way we can avoid manually refetching from here.

Reply:
That's what I originally had, but then it leads back to the same issue with the table flickering an incorrect value when it's loaded, since the first one will be from the cache and then it'll grab it from the network. Also tried network-only, and whilst that worked, I didn't really like the refetching every single time, this seemed much cleaner. 🧹